nevergrad/optimization/optimizerlib.py

```diff
@@ -2421,3 +2421,57 @@ def recommend(self) -> p.Parameter:
 @registry.register
 class NGOpt(NGOpt10):
     pass
+
+
+class _MSR(CM):
+    """This code applies multiple copies of NGOpt with random weights for the different objective functions.
+
+    Variants dedicated to multiobjective optimization by multiple singleobjective optimization.
+    """
+
+    def __init__(
+        self,
+        parametrization: IntOrParameter,
+        budget: tp.Optional[int] = None,
+        num_workers: int = 1,
+        num_single_runs: int = 9,
+        base_optimizer: base.OptCls = NGOpt,
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        assert budget is not None
```

Comment:
you should remove this, and propagate budget = None to the suboptimizers is budget is None

Reply:
Done.